src/index.d.ts

```diff
@@ -0,0 +1,1130 @@
+}
+
+export function getFileInfo(
+  filePath: string,
+  options?: FileInfoOptions
+): Promise<FileInfoResult>;
+
+/**
+ * Returns an object representing the parsers, languages and file types Prettier supports for the current version.
+ */
+export function getSupportInfo(): Promise<SupportInfo>;
+
+/**
+ * `version` field in `package.json`
+ */
+export const version: string;
+
+// https://github.com/prettier/prettier/blob/next/src/utils/public.js
+export namespace util {
+  interface SkipOptions {
+    backwards?: boolean | undefined;
+  }
+
+  type Quote = "'" | '"';
+
+  function getMaxContinuousCount(text: string, searchString: string): number;
+
+  function getStringWidth(text: string): number;
+
+  function getAlignmentSize(
+    text: string,
+    tabWidth: number,
+    startIndex?: number | undefined
+  ): number;
+
+  function getIndentSize(value: string, tabWidth: number): number;
+
+  function skipNewline(
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions | undefined
+  ): number | false;
+
+  function skipInlineComment(
+    text: string,
+    startIndex: number | false
+  ): number | false;
+
+  function skipTrailingComment(
+    text: string,
+    startIndex: number | false
+  ): number | false;
+
+  function skipTrailingComment(
+    text: string,
+    startIndex: number | false
+  ): number | false;
+
+  function hasNewline(
+    text: string,
+    startIndex: number,
+    options?: SkipOptions | undefined
+  ): boolean;
+
+  function hasNewlineInRange(
+    text: string,
+    startIndex: number,
+    endIndex: number
+  ): boolean;
+
+  function hasSpaces(
+    text: string,
+    startIndex: number,
+    options?: SkipOptions | undefined
+  ): boolean;
+
+  function getNextNonSpaceNonCommentCharacter(
+    text: string,
+    startIndex: number
+  ): string;
+
+  function makeString(
+    rawText: string,
+    enclosingQuote: Quote,
+    unescapeUnnecessaryEscapes?: boolean | undefined
+  ): string;
+
+  function skip(
+    characters: string | RegExp
+  ): (
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions
+  ) => number | false;
+
+  const skipWhitespace: (
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions
+  ) => number | false;
+
+  const skipSpaces: (
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions
+  ) => number | false;
+
+  const skipToLineEnd: (
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions
+  ) => number | false;
+
+  const skipEverythingButNewLine: (
+    text: string,
+    startIndex: number | false,
+    options?: SkipOptions
+  ) => number | false;
+
+  function addLeadingComment(node: any, comment: any): void;
+
+  function addDanglingComment(node: any, comment: any, marker: any): void;
+
+  function addTrailingComment(node: any, comment: any): void;
+}
+
+// https://github.com/prettier/prettier/blob/next/src/document/index.js
+export namespace doc {
```

Comment:
Is it possible to move this to `src/document/index.js`? Since we bundle `doc.js` too.